Author: Paul Brossier <piem@piem.org>
Description: add hurd glue
diff --git a/src/d_array.c b/src/d_array.c
index 11a55b5..b3b62f4 100644
--- a/src/d_array.c
+++ b/src/d_array.c
@@ -523,6 +523,7 @@ static void tabread4_tilde_setup(void)
 #endif
 
 #if defined(__unix__) || defined(__APPLE__)
+#include <endian.h>
 #if !defined(BYTE_ORDER) || !defined(LITTLE_ENDIAN)                         
 #error No byte order defined                                                    
 #endif                                                                          
diff --git a/src/d_osc.c b/src/d_osc.c
index 8336dd3..bd4198b 100644
--- a/src/d_osc.c
+++ b/src/d_osc.c
@@ -35,6 +35,7 @@
 #endif
 
 #if defined(__unix__) || defined(__APPLE__)
+#include <endian.h>
 #if !defined(BYTE_ORDER) || !defined(LITTLE_ENDIAN)                         
 #error No byte order defined                                                    
 #endif                                                                          
diff --git a/src/configure b/src/configure
index 8c8275e..b7867a9 100755
--- a/src/configure
+++ b/src/configure
@@ -7354,10 +7354,16 @@ fi
       LDFLAGS="$LDFLAGS -static"
     fi
     EXT=pd_linux
-    CPPFLAGS="-DDL_OPEN -DPA_USE_OSS -DUNIX -DUNISTD\
-        -DUSEAPI_OSS \
+    CPPFLAGS="-DDL_OPEN -DUNIX -DUNISTD\
         -fno-strict-aliasing"
-    SYSSRC="s_midi_oss.c s_audio_oss.c"
+    dnl No OSS on hurd
+    if test `uname -s` = "GNU";
+    then
+        SYSSRC="s_midi_dummy.c"
+    else
+        SYSSRC="s_midi_oss.c s_audio_oss.c"
+        CPPFLAGS=$CPPFLAGS" -DPA_USE_OSS -DUSEAPI_OSS"
+    fi
     if test x$alsa = "xyes";
     then
         SYSSRC=$SYSSRC" s_audio_alsa.c s_audio_alsamm.c s_midi_alsa.c"
diff --git a/src/configure.in b/src/configure.in
index bc5d5f7..e640c82 100644
--- a/src/configure.in
+++ b/src/configure.in
@@ -195,10 +195,16 @@ dnl This should be fixed so Pd can use ALSA shared libraries where appropriate.
       LDFLAGS="$LDFLAGS -static"
     fi 
     EXT=pd_linux
-    CPPFLAGS="-DDL_OPEN -DPA_USE_OSS -DUNIX -DUNISTD\
-        -DUSEAPI_OSS \
+    CPPFLAGS="-DDL_OPEN -DUNIX -DUNISTD\
         -fno-strict-aliasing"
-    SYSSRC="s_midi_oss.c s_audio_oss.c"
+    dnl No OSS on hurd
+    if test `uname -s` = "GNU";
+    then
+        SYSSRC="s_midi_dummy.c"
+    else
+        SYSSRC="s_midi_oss.c s_audio_oss.c"
+        CPPFLAGS=$CPPFLAGS" -DPA_USE_OSS -DUSEAPI_OSS"
+    fi
     if test x$alsa = "xyes";
     then
         SYSSRC=$SYSSRC" s_audio_alsa.c s_audio_alsamm.c s_midi_alsa.c"
diff --git a/src/s_stuff.h b/src/s_stuff.h
index 0fb2497..a997370 100644
--- a/src/s_stuff.h
+++ b/src/s_stuff.h
@@ -200,6 +200,10 @@ void sys_setvirtualalarm( void);
 #define API_DEFAULT API_SGI
 #define API_DEFSTRING "SGI Digital Media"
 #endif
+#ifdef __GNU__
+#define API_DEFAULT API_JACK
+#define API_DEFSTRING "Jack audio connection kit"
+#endif
 #define DEFAULTAUDIODEV 0
 
 #define MAXAUDIOINDEV 4
--- /dev/null	2007-12-29 19:47:02.000000000 +0000
+++ b/src/s_midi_dummy.c	2009-12-22 05:55:53.000000000 +0000
@@ -0,0 +1,32 @@
+/* Copyright (c) 1997-1999 Guenter Geiger, Miller Puckette, Larry Troxler,
+* Winfried Ritsch, Karl MacMillan, and others.
+* For information on usage and redistribution, and for a DISCLAIMER OF ALL
+* WARRANTIES, see the file, "LICENSE.txt," in this distribution.  */
+
+/* Dummy midi driver for platforms without midi drivers */
+
+void sys_do_open_midi(int nmidiin, int *midiinvec,
+    int nmidiout, int *midioutvec)
+{
+}
+
+void sys_putmidimess(int portno, int a, int b, int c)
+{
+}
+
+void sys_putmidibyte(int portno, int byte)
+{
+}
+
+void sys_poll_midi(void)
+{
+}
+
+void sys_close_midi()
+{
+}
+
+void midi_getdevs(char *indevlist, int *nindevs,
+    char *outdevlist, int *noutdevs, int maxndev, int devdescsize)
+{
+}
