Author: IOhannes m zmölnig
Description: make ALSA the default audio API on linux
--- puredata.orig/src/s_stuff.h
+++ puredata/src/s_stuff.h
@@ -204,10 +204,14 @@
 #define API_DEFAULT API_DUMMY
 #define API_DEFSTRING "dummy audio"
 #else
-#if defined(__linux__) || defined(__FreeBSD_kernel__)
+#if defined(__FreeBSD_kernel__) 
 # define API_DEFAULT API_OSS
 # define API_DEFSTRING "OSS"
 #endif
+#if defined(__linux__) 
+# define API_DEFAULT API_ALSA
+# define API_DEFSTRING "ALSA"
+#endif
 #if defined(_WIN32) || defined(__CYGWIN__)
 # define API_DEFAULT API_MMIO
 # define API_DEFSTRING "MMIO"
