Author: IOhannes m zmölnig
Description: fix paths to some extra/objects
--- puredata.orig/doc/7.stuff/soundfile-tools/4.looper.pd
+++ puredata/doc/7.stuff/soundfile-tools/4.looper.pd
@@ -74,7 +74,7 @@
 #X obj 601 61 loadbang;
 #X msg 598 91 \; transposition 0 \; looplength 0 \; startpoint 0 \;
 smoothing 0;
-#X obj 236 263 ../../../extra/loop~;
+#X obj 236 263 loop~;
 #X obj 285 233 * 441;
 #X obj 284 307 *~;
 #X obj 133 41 loadbang;
--- puredata.orig/doc/7.stuff/soundfile-tools/5.reverb.pd
+++ puredata/doc/7.stuff/soundfile-tools/5.reverb.pd
@@ -42,7 +42,7 @@
 #N canvas 0 0 632 395 audio-transformation 0;
 #X obj 101 49 inlet~;
 #X obj 105 268 outlet~;
-#X obj 101 148 ../../../extra/rev1~ xxx;
+#X obj 101 148 rev1~ xxx;
 #X obj 339 79 r revgain;
 #X obj 338 102 dbtorms;
 #X obj 338 130 pack 0 50;
