Author: IOhannes m zmölnig
Description: Fixing spelling errors
--- puredata.orig/src/m_glob.c
+++ puredata/src/m_glob.c
@@ -66,7 +66,7 @@
             post("warning: file format (%g) newer than this version (%g) of Pd",
                 f, PD_MAJOR_VERSION + 0.01*PD_MINOR_VERSION);
         else if (warned < 2)
-            post("(... more file format messsages suppressed)");
+            post("(... more file format messages suppressed)");
         warned++;
     }
 }
